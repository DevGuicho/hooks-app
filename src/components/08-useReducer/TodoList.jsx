import React from "react";
import TodoListItem from "./TodoListItem";

const TodoList = ({ todos, handleDelete, handleToggle }) => {
  return (
    <div className="col-8">
      <table className="table table-striped table-hover">
        <thead>
          <tr>
            <th scope="col">#</th>
            <th scope="col">Description</th>
            <th scope="col">Actions</th>
            <th scope="col">Complete</th>
          </tr>
        </thead>
        <tbody>
          {todos.map((todo, i) => (
            <TodoListItem
              key={todo.id}
              todo={todo}
              i={i}
              handleDelete={handleDelete}
              handleToggle={handleToggle}
            />
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default TodoList;
