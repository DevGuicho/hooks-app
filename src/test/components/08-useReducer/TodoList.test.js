import { shallow } from "enzyme";
import TodoList from "../../../components/08-useReducer/TodoList";
import { demoTodos } from "../../fixtures/demoTodos";

describe("Pruebas en <TodoList />", () => {
  const handleToggle = jest.fn();
  const handleDelete = jest.fn();
  const wrapper = shallow(
    <TodoList
      handleDelete={handleDelete}
      handleToggle={handleToggle}
      todos={demoTodos}
    />
  );
  test("debe mostrar correctamente el componente", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("Debe tener dos <TodoListItem />", () => {
    expect(wrapper.find("TodoListItem").length).toBe(demoTodos.length);
    expect(wrapper.find("TodoListItem").at(0).prop("handleDelete")).toEqual(
      expect.any(Function)
    );
  });
});
