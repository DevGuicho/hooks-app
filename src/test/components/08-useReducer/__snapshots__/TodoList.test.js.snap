// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoList /> debe mostrar correctamente el componente 1`] = `
<div
  className="col-8"
>
  <table
    className="table table-striped table-hover"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          #
        </th>
        <th
          scope="col"
        >
          Description
        </th>
        <th
          scope="col"
        >
          Actions
        </th>
        <th
          scope="col"
        >
          Complete
        </th>
      </tr>
    </thead>
    <tbody>
      <TodoListItem
        handleDelete={[MockFunction]}
        handleToggle={[MockFunction]}
        i={0}
        key="1"
        todo={
          Object {
            "desc": "Hola Mundo",
            "done": false,
            "id": 1,
          }
        }
      />
      <TodoListItem
        handleDelete={[MockFunction]}
        handleToggle={[MockFunction]}
        i={1}
        key="2"
        todo={
          Object {
            "desc": "Hola Mundo 2",
            "done": false,
            "id": 2,
          }
        }
      />
    </tbody>
  </table>
</div>
`;
